django/db/models/query.py

```diff
@@ -1190,6 +1190,8 @@ def update(self, **kwargs):
         # Inline annotations in order_by(), if possible.
         new_order_by = []
         for col in query.order_by:
+            if isinstance(col, str) and col.startswith('-'):
+                col = col[1:]
             if annotation := query.annotations.get(col):
                 if getattr(annotation, "contains_aggregate", False):
                     raise exceptions.FieldError(
```

Comment:
The code is not technically correct as it strips the descending specifier that should be part of the `order_by` and is important on some databases such as MySQL (hence the failing test).

The logic here should determine if a descending order is specified and if it's the case
1. If it's referencing an annotation `annotation.desc()` should be added to `new_order_by`
2. If it's not referencing an annotation then `col` should be preserved as is.

Reply:
You're right, I just checked for errors and didn't consider the side effects. I have modified the code. Thanks!

Reply:
Could you also add a new test to `MySQLUpdateOrderByTest` that copies `test_order_by_update_on_unique_constraint_annotation` but annotates an alias to the field and does the descending bits in order by

```
def test_order_by_update_on_unique_constraint_annotation_desc(self):
        updated = (
            UniqueNumber.objects.annotate(number_annotation=F("number"))
            .order_by("-number_annotation")
            .update(number=F("number") + 1)
        )
        self.assertEqual(updated, 2)
```

Reply:
Ok, I've added this test